Sources/CNIOLinux/shim.c

```diff
@@ -150,4 +150,21 @@ size_t CNIOLinux_CMSG_SPACE(size_t payloadSizeBytes) {
 
 const int CNIOLinux_SO_TIMESTAMP = SO_TIMESTAMP;
 const int CNIOLinux_SO_RCVTIMEO = SO_RCVTIMEO;
+
+int CNIOLinux_supports_udp_segment() {
+    #ifndef UDP_SEGMENT
```

Comment:
This guard isn't enough: we need to compile out the function below if `UDP_SEGMENT` isn't defined, or the compiler will still fail.

Reply:
Yes, the preprocessor doesn't skip the rest of the code, and the compiler doesn't stop parsing after `return`. [Godbolt](https://godbolt.org/z/dxvjKvTrc).